PowerShell/ScubaGear/Modules/Providers/ExportAADProvider.psm1

```diff
@@ -1,3 +1,51 @@
+# Many of the commandlets can be replaced with direct API access, but we are starting the transition with the ones
+# below because they have slow imports that affect performance more than the others. Some commandlets are fast
+# and there is no obvoius performance advantage to using the API beyond maybe batching.
+$GraphEndpoints = @{
+    "Get-MgBetaRoleManagementDirectoryRoleEligibilityScheduleInstance" = "/beta/roleManagement/directory/roleEligibilityScheduleInstances"
+    "Get-MgBetaRoleManagementDirectoryRoleAssignmentScheduleInstance" = "/beta/roleManagement/directory/roleAssignmentScheduleInstances"
+    "Get-MgBetaIdentityGovernancePrivilegedAccessGroupEligibilityScheduleInstance" = "/beta/identityGovernance/privilegedAccess/group/eligibilityScheduleInstances"
+    "Get-MgBetaPrivilegedAccessResource" = "/beta/privilegedAccess/aadGroups/resources"
+}
+
+function Invoke-GraphDirectly {
+    param (
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $commandlet,
+
+        [System.Collections.Hashtable]
+        $queryParams
+    )
+
+    Write-Debug "Replacing Cmdlet: $commandlet"
+    try {
+        $endpoint = $GraphEndpoints[$commandlet]
+    } catch {
+        Write-Error "The commandlet $commandlet can't be used with the Invoke-GraphDirectly function yet."
+    }
+
+    if ($queryParams) {
+        <#
+          If query params are passed in, we need to augment the endpoint URI to include the params. Paperwork below.
+          We can't use string formatting easily because the graph API expects parameters
+          that are prefixed with the dollar symbol. Maybe someone smarter than me can figure that out.
+        #>
+        $q = [System.Web.HttpUtility]::ParseQueryString([string]::Empty)
+        foreach ($item in $queryParams.GetEnumerator()) {
+            $q.Add($item.Key, $item.Value)
+        }
+        $endpoint = "https://example.com" + $endpoint
+        $uri = [System.UriBuilder]$endpoint
+        $uri.Query = $q.ToString()
+        $endpoint = $uri.Path + $uri.Query
```

Comment:
This seems off.  

You should be able to build the Uri using host, query string, scheme, etc and then use uri.ToString():

https://learn.microsoft.com/en-us/dotnet/api/system.uribuilder?view=net-8.0

Reply:
I will change it.

Reply:
It works passing all 5 params interactively in powershell, but refuses the 5th parameter (query string) when I add this to the Scuba code. I'm at a loss for this goofy behavior, so I settled on passing all of them in and concat'ing the query string. I will push the changes so everyone can see them.